qiskit/pulse/instructions/frequency.py

```diff
@@ -60,3 +60,32 @@ def channel(self) -> PulseChannel:
         scheduled on.
         """
         return self._channel
+
+
+class ShiftFrequency(Instruction):
+    r"""Shift the channel frequency away from the LO frequency."""
+    def __init__(self, frequency: float,
+                 channel: Optional[PulseChannel],
```

Comment:
Channel should not be optional. It's optional for the existing instructions because we have to deprecate the command style implementation

Reply:
Fixed in [e8dd486](https://github.com/Qiskit/qiskit-terra/pull/4390/commits/e8dd48689b1a2502e19f59785619a63f4de82d25).